### PR TITLE
Key 686: process empty arrays

### PR DESCRIPTION
**Pull request by zxan1285**
## ✏️ Changes
 `auth0-source-control-extension-tools` updated to v3.2.0, which allows to process empty arrays t oremove records form the tenant. Allow passing `undefined` assets instead of converting them into empty arrays to be able to ignore items.
  
## 🔗 References
  Github issue: #74 
  
## 🎯 Testing
✅ This change has been tested locally
✅ This change has unit test coverage

### CODE REVIEW

**Comment by stevezau** [comment]
😄

File: test/context/yaml/context.test.js [hunk 1]
```diff
@@ -18,11 +18,11 @@ describe('#YAML context validation', () => {
...
-    expect(context.assets.rules).to.deep.equal([]);
-    expect(context.assets.databases).to.deep.equal([]);
-    expect(context.assets.pages).to.deep.equal([]);
-    expect(context.assets.clients).to.deep.equal([]);
-    expect(context.assets.resourceServers).to.deep.equal([]);
+    expect(context.assets.rules).to.deep.equal(undefined);
+    expect(context.assets.databases).to.deep.equal(undefined);
+    expect(context.assets.pages).to.deep.equal(undefined);
+    expect(context.assets.clients).to.deep.equal(undefined);
+    expect(context.assets.resourceServers).to.deep.equal(undefined);
```
Comment (fyockm) [inline test/context/yaml/context.test.js]: we are missing tests for some resources, specifically `clientGrants`, `connections` and `rulesConfigs`

**Comment by twistedstream** [comment]
Nice work guys!

I noticed that this new updated clears all entities with empty arrays _except_ for `databases`. Was that by design?

cc @zxan1285 @stevezau @fyockm

**Comment by stevezau** [comment]
@twistedstream think you are right.. https://github.com/zxan1285/auth0-deploy-cli/blob/4258fc1d70fe36b778a5cec863b12a8babd9fc4b/src/context/yaml/handlers/databases.js#L9 needs to be an array

@zxan1285 ??

**Comment by zxan1285** [comment]
Yes, my mistake. But `deploy-cli` is fine. I missed database handler in `source-control-tools`. Here: https://github.com/auth0-extensions/auth0-source-control-extension-tools/pull/61

**Pull request merged**